#! /bin/sh /usr/share/dpatch/dpatch-run
## 23-PortableGroup-exceptions-propagation.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Remove exception specification preventing exceptions from
## DP: being propagated to caller.
## DP: See ChangeLog entry `Thu Sep 15 14:04:28 2005  Ossama Othman  <ossama@dre.vanderbilt.edu>'

@DPATCH@

--- ACE_wrappers/TAO/orbsvcs/orbsvcs/PortableGroup/PortableGroup_Request_Dispatcher.h-
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/PortableGroup/PortableGroup_Request_Dispatcher.h
@@ -33,12 +33,13 @@
  * @brief A class that strategizes the request dispatching procedure.
  *
  */
-class TAO_PortableGroup_Export PortableGroup_Request_Dispatcher :
-  public TAO_Request_Dispatcher
+class TAO_PortableGroup_Export PortableGroup_Request_Dispatcher
+  : public TAO_Request_Dispatcher
 {
   friend class TAO_GOA;
 
 public:
+
   /// Destructor.
   virtual ~PortableGroup_Request_Dispatcher (void);
 
@@ -48,8 +49,7 @@
   virtual void dispatch (TAO_ORB_Core *orb_core,
                          TAO_ServerRequest &request,
                          CORBA::Object_out forward_to
-                         ACE_ENV_ARG_DECL)
-    ACE_THROW_SPEC ((CORBA::SystemException));
+                         ACE_ENV_ARG_DECL);
 
 private:
 
@@ -58,6 +58,7 @@
 
   /// Registry for all of the group (multicast) acceptors.
   TAO_PortableGroup_Acceptor_Registry acceptor_registry_;
+
 };
 
 #include /**/ "ace/post.h"
--- ACE_wrappers/TAO/orbsvcs/orbsvcs/PortableGroup/PortableGroup_Request_Dispatcher.cpp-
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/PortableGroup/PortableGroup_Request_Dispatcher.cpp
@@ -22,7 +22,6 @@
                                             TAO_ServerRequest &request,
                                             CORBA::Object_out forward_to
                                             ACE_ENV_ARG_DECL)
-  ACE_THROW_SPEC ((CORBA::SystemException))
 {
   // Check if the request contains a tagged profile
   if (request.profile ().discriminator () == GIOP::ProfileAddr)
